core/libraries/rest_api/ModelDataTranslator.php

```diff
@@ -382,223 +385,395 @@ public static function prepareConditionsQueryParamsForModels(
+     */
+    private static function determineFieldAndTimezone(EEM_Base $model, $query_param_key)
+    {
+        $query_param_sans_stars = ModelDataTranslator::removeStarsAndAnythingAfterFromConditionQueryParamKey(
+            $query_param_key
+        );
+        $is_gmt_datetime_field = false;
+        $field = ModelDataTranslator::deduceFieldFromQueryParam(
+            $query_param_sans_stars,
+            $model
+        );
+        // double-check is it a *_gmt field?
+        if (! $field instanceof EE_Model_Field_Base
+            && ModelDataTranslator::isGmtDateFieldName($query_param_sans_stars)
+        ) {
+            // yep, take off '_gmt', and find the field
+            $query_param_key = ModelDataTranslator::removeGmtFromFieldName($query_param_sans_stars);
+            $field = ModelDataTranslator::deduceFieldFromQueryParam(
+                $query_param_key,
+                $model
+            );
+            $timezone = 'UTC';
+            $is_gmt_datetime_field = true;
+        } elseif ($field instanceof EE_Datetime_Field) {
+            // so it's not a GMT field. Set the timezone on the model to the default
+            $timezone = \EEH_DTT_Helper::get_valid_timezone_string();
+        } else {
+            // just keep using what's already set for the timezone
+            $timezone = $model->get_timezone();
+        }
+        return array(
+            $field,
+            $query_param_key,
+            $query_param_sans_stars,
+            $timezone,
+            $is_gmt_datetime_field
+        );
+    }
+
+    /**
+     * Given a ton of input, determines the value to use for the models.
+     * @since $VID:$
+     * @param EEM_Base $model
+     * @param EE_Model_Field_Base $field
+     * @param $requested_version
+     * @param $timezone
+     * @param $query_param_key
+     * @param $query_param_value
+     * @param bool $writing
+     * @return array|null
+     * @throws DomainException
+     * @throws EE_Error
+     * @throws RestException
+     */
+    private static function determineConditionsQueryParameterValue(
+        EEM_Base $model,
+        EE_Model_Field_Base $field,
+        $requested_version,
+        $timezone,
+        $query_param_key,
+        $query_param_value,
+        $writing = false
+    ) {
+        if (! $writing && is_array($query_param_value)) {
+            if (! \EEH_Array::is_array_numerically_and_sequentially_indexed($query_param_value)) {
+                if (is_array($query_param_value)
+                    && count($query_param_value) === 1
+                    && array_key_exists(
+                        key($query_param_value),
+                        $model->valid_operators()
+                    )
+                ) {
+                    $sub_array_value =  reset($query_param_value);
+                    $sub_array_key = key($query_param_value);
+                    if (is_array($sub_array_value) && defined('EE_REST_API_DEBUG_MODE') && EE_REST_API_DEBUG_MODE) {
```

Comment:
kk I added the method to `EED_Core_Rest_Api` so other classes can use it, although I suppose it could be on more SOLID class... not sure which. I could live with it on `EED_Core_Rest_Api` too